dynamic/clientset/clientset.go

```diff
@@ -175,3 +175,32 @@ func (rc *ResourceClient) RemoveFinalizer(orig *unstructured.Unstructured, name
 		return true
 	})
 }
+
+// AtomicStatusUpdate is similar to AtomicUpdate, except that it updates status.
+func (rc *ResourceClient) AtomicStatusUpdate(resourceMap *dynamicdiscovery.ResourceMap, parentResource *dynamicdiscovery.APIResource, orig *unstructured.Unstructured, update func(obj *unstructured.Unstructured) bool) (result *unstructured.Unstructured, err error) {
```

Comment:
We shouldn't need to add `resourceMap` or `parentResource`. This should have the same arguments as `AtomicUpdate`.